server/src/main/java/org/elasticsearch/transport/NoSeedNodeLeftException.java

```diff
@@ -9,24 +9,30 @@
 package org.elasticsearch.transport;
 
 import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.ElasticsearchWrapperException;
 import org.elasticsearch.common.io.stream.StreamInput;
 
 import java.io.IOException;
 
 /**
  * Thrown after completely failing to connect to any node of the remote cluster.
  */
-public class NoSeedNodeLeftException extends ElasticsearchException {
+public class NoSeedNodeLeftException extends ElasticsearchException implements ElasticsearchWrapperException {
```

Comment:
This change means that `ExceptionsHelper#unwrapCause` will now unwrap this exception, which might change the behaviour of retries in CCR:

https://github.com/elastic/elasticsearch/blob/43b69b6f704da5b0a41f1785635f2c894bd9d2e5/x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java#L626-L640

That's ok if the inner exception is a `ConnectTransportException`, but it looks from the code in `SniffConnectionStrategy` as if we also expect perhaps an `IOException` or an `IllegalStateException` here.

Reply:
Thanks for the pointer. I removed it. It was added so that the user level error is 401 instead of 500. But on second thought, 500 is probably more reasonable because authentication does *not* fail on REST request and the issue is probably a configuration issue which cannot fixed by end-user who issues the request.